sbi/samplers/importance/importance_sampling.py

```diff
@@ -0,0 +1,128 @@
+from math import sqrt
+from typing import Tuple
+
+import torch
+from torch import Tensor
+
+
+def importance_sample(
+    potential_fn,
+    proposal,
+    num_samples: int = 1,
+) -> Tuple[Tensor, Tensor]:
+    """Returns samples from proposal and log(importance weights).
+
+    Args:
+        potential_fn: Unnormalized potential function.
+        proposal: Proposal distribution with `.sample()` and `.log_prob()` methods.
+        num_samples: Number of samples to draw.
+
+    Returns:
+        Samples and logarithm of importance weights.
+    """
+    samples = proposal.sample((num_samples,))
+
+    potential_logprobs = potential_fn(samples)
+    proposal_logprobs = proposal.log_prob(samples)
+    log_importance_weights = potential_logprobs - proposal_logprobs
+
+    return samples, log_importance_weights
+
+
+def exponentiate_weights(log_weights: Tensor) -> Tensor:
+    """Subtracts the maximum of the `log_weights` and then exponentiates them.
+
+    Args:
+        log_weights: Logarithm of the importance weights.
+
+    Returns:
+        Tensor: Importance weights.
+    """
+    log_weights = log_weights[torch.isfinite(log_weights)]
+    logweights_max = log_weights.max()
+    weights = torch.exp(log_weights - logweights_max)
+    return weights
+
+
+def largest_weight_indices(weights: Tensor) -> Tensor:
+    """Returns the indizes of the largest weights.
+
+    Args:
+        weights: Importance weights.
+
+    Returns:
+        Tensor: The indices of the largest importance weights.
+    """
+    number_of_weights = int(min(len(weights) / 5, 3 * sqrt(len(weights))))
```

Comment:
where do these numbers come from? is there a reference? Maybe better to have them as arguments? In any case, it would be great to have more explanation here.